frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch.ts

```diff
@@ -32,17 +32,30 @@ const consoleFetchCommon = async (
   method: string = 'GET',
   options: RequestInit = {},
   timeout?: number,
+  isFullResponse?: boolean,
 ) => {
   const headers = getConsoleRequestHeaders();
+  const defaultIsFullResponse = typeof isFullResponse !== 'boolean' ? false : isFullResponse;
+
   // Pass headers last to let callers to override Accept.
   const allOptions = _.defaultsDeep({ method }, options, { headers });
   const response = await consoleFetch(url, allOptions, timeout);
   const text = await response.text();
   const isPlainText = response.headers.get('Content-Type') === 'text/plain';
   if (!text) {
-    return isPlainText ? '' : {};
+    return isPlainText ? response : {};
```

Comment:
This doesn't seem right. If there is no text in the response body, should we always return the full response?